src/harness/reference_models/interference/aggregate_interference.py

```diff
@@ -0,0 +1,366 @@
+
+
+def convertAndSumInterference(cbsd_interference_list):
+  """Converts interference in dBm to mW to calculate aggregate interference"""
+  
+  interferences_in_dbm = np.array(cbsd_interference_list)
+  return np.sum(interf.dbToLinear(interferences_in_dbm))
+
+
+def aggregateInterferenceForPoint(protection_point, channels, low_freq, high_freq, 
+      grant_objects, fss_info, esc_antenna_info, protection_ent_type, 
+      region_type="SUBURBAN"):
+  """Algorithm to compute aggregate interference for protection point
+  
+  This method is invoked to calculate aggregate interference for FSS(Co-channel 
+  passband) and ESC protection points.It is also invoked for protection points 
+  within PPA and GWPZ protection areas.
+
+  Args:
+   protection_point: Protection point inside protection area
+   channels: Channels over which reference incumbent needs protection
+   low_freq: Low frequency of protection constraint (Hz)
+   high_freq: High frequency of protection constraint (Hz)
+   grant_objects: A list of grant requests, each one being a namedtuple
+                  containing grant information
+   fss_info: Contains fss point and antenna information
+   esc_antenna_info: Contains information on ESC antenna height, azimuth,
+                     gain and pattern gain 
+   protection_ent_type: An enum member of class ProtectedEntityType
+   region_type: Region type of the protection point
+   Returns:
+      Aggregate interference for a protection constraint(mW)
+  """
+
+  # Get all the grants inside neighborhood of the protection entity
+  grants_inside = interf.findGrantsInsideNeighborhood(
+                    grant_objects, protection_point, protection_ent_type)
+
+  if len(grants_inside) > 0:
+    for channel in channels:
+      # Get protection constraint over 5MHz channel range
+      protection_constraint = interf.ProtectionConstraint(latitude=protection_point[0],
+                                longitude=protection_point[1], low_frequency=channel[0],
+                                high_frequency=channel[1], entity_type=protection_ent_type)
+
+      # Identify CBSD grants overlapping with frequency range of the protection entity 
+      # in the neighborhood of the protection point and channel 
+      neighborhood_grants = interf.findOverlappingGrants(
+                              grants_inside, protection_constraint)
+
+      if not neighborhood_grants:
+        continue
+     
+      cbsd_interference_list = []
+
+      for grant in neighborhood_grants:
+        cbsd_interference_list.append(interf.computeInterference(
+          grant, grant.max_eirp, protection_constraint, fss_info, 
+          esc_antenna_info, region_type))
+
+      aggr_interference = convertAndSumInterference(cbsd_interference_list)
+
+      # Update aggregate interference for a protection constraint
+      aggregate_interference.SetAggregateInterferenceInfo(protection_point[0],
+            protection_point[1], aggr_interference)
```

Comment:
With current architecture where this function update the aggregate_interference object, 
it should really take it as an input instead of modifying the global variable: first param as is the convention for such thing, like an object update its self.
And to make it clear it is actually a function with side effects, could be renamed:
updateAggregateInterferenceForPoint(agg_interference, ....)

Reply:
passed   AggergateInterferenceOutputFormat object as function parameter